### PR TITLE
Make notification for Logger.error configurable

### PR DESCRIPTION
**Pull request by yukster**
Follow up to upstream issue #315; adds a configuration key for controlling whether Logger.error sends a Honeybadger notification. Default is false.

### CODE REVIEW

**Review by effektzwm — APPROVED** [review]

**Comment by TraceyOnim** [comment]
Tested it, this looks good

**Review by TraceyOnim — APPROVED** [review]

File: test/honeybadger/logger_test.exs [hunk 1]
```diff
@@ -146,11 +146,22 @@ defmodule Honeybadger.LoggerTest do
...
 
-    assert_receive {:api_request, %{"breadcrumbs" => breadcrumbs}}
-    assert List.first(breadcrumbs["trail"])["metadata"]["message"] == "Error-level log"
+      assert_receive {:api_request, %{"breadcrumbs" => breadcrumbs}}
+      assert List.first(breadcrumbs["trail"])["metadata"]["message"] == "Error-level log"
+    end)
+  end
+
+  test "ignores error-level log when disabled" do
+    with_config([notify_for_error_logs: true], fn ->
```
[the remark this answers is not in the document]
Comment (yukster) [reply test/honeybadger/logger_test.exs]: Duh, whoops. Fixed. Thanks!

**Comment by yukster** [comment]
@sorentwo what do you think?

**Comment by sorentwo** [comment]
Admittedly, I'm out of the loop a little with all the recent issues around this. However, isn't the purpose of `use_logger` to attach a logger that sends notifications on error events? I'd expect the same functionality with `use_logger: false` that we'd get with the addition of this new option.

Chances are that I'm missing something! Is there a reason that `use_logger: false` is undesirable?

**Comment by yukster** [comment]
Hi @sorentwo! The backstory is that `use_logger` is how Honeybadger gets notifications for any non-web Elixir/Erlang crashes. At least that's what appeared to happen in my testing. If I set it to false then GenServer or Task exceptions did not get reported (and they did with it set to true). The discussion was in this issue: https://github.com/honeybadger-io/honeybadger-elixir/issues/276#issuecomment-884378152

**Comment by sorentwo** [comment]
@yukster Ok, I see now. That's what the investigation around SASL vs regular error logging was about. Yes, to get notifications about anything built on the "gen" behaviors we need SASL logging, but we don't necessarily want reports for things reported at the `:error` level.

Perhaps a different option name is in order? There are other error log levels in recent OTP versions ([critical, alert, emergency](http://erlang.org/doc/apps/kernel/logger_chapter.html#log-level)) and the name doesn't make it obvious which levels are reported.

What if `use_logger` is expanded to have an option like `:sasl_only`, or accept a level rather than a boolean? Or, to keep the options totally separate you can flip this option into something like `sasl_logging_only`. WDYT?

**Comment by yukster** [comment]
@sorentwo hm yeah, good point about varying log levels. Just renaming the config seemed more straightforward to me so I went ahead and did that. What do you think?

**Review by sorentwo — APPROVED** [review]
Looks great with the clearer name!

**Comment by yukster** [comment]
Awesome, thanks @sorentwo, @TraceyOnim, and @effektzwm!

**Pull request merged**